packages/wrangler/src/config/index.ts

```diff
@@ -130,6 +134,31 @@ export function findWranglerToml(
 	return findUpSync(`wrangler.toml`, { cwd: referencePath });
 }
 
+/**
+ * Pages projects require the `pages_build_output_dir` to be specififed in
+ * their `wrangler.toml` file. This function returns the configuration file
+ * path, only if the configuration is valid for Pages.
+ */
+export function findPagesWranglerToml(
+	referencePath: string = process.cwd()
+): string | undefined {
+	const configPath = findWranglerToml(referencePath);
+
+	if (!configPath) return;
+	if (!isPagesWranglerToml(configPath)) return;
+
+	return configPath;
+}
+
+export function isPagesWranglerToml(
+	configPath: string | undefined
+): boolean | undefined {
+	if (!configPath) return;
+
+	const rawConfig = parseTOML(readFileSync(configPath), configPath);
+	return _isPagesConfig(rawConfig);
+}
+
```

Comment:
As above—could you adapt this to use #5449?

Reply:
done